### DISCUSSION TITLE
Dynamically import SCSS

### DISCUSSION THREAD

**Discussion by joulev** [vercel/next.js#31432] [Help] [answered]
Now I have two files `variables.dark.scss` and `variables.light.scss`, so now I can choose my app theme by editing the `@import` statement in my `global.scss`.

If I want to, I can also import these variable files to my component SCSS module files, so I can basically choose the theme of anything on my app individually.

Now I would like to find a way to change the theme with app properties instead of by manually editing SCSS files. However I realised that dynamic import does not work for CSS/SCSS.

For example, if I try to make two files `myComponent.light.module.scss` and `myComponent.dark.module.scss`, and import it with something like this

```jsx
const { styles } = import(`./myComponent.{props.dark ? "dark" : "light"}.module.scss`);
```

I (expectedly) received an error.

```
error - ./components/myComponent.dark.module.scss
CSS Modules cannot be imported from within node_modules.
Read more: https://nextjs.org/docs/messages/css-modules-npm
```

What should I do for this problem?

**Comment by SamuelColeMorgan** [comment, marked as answer]
<img width="419" alt="Screen Shot 2021-11-15 at 9 04 49 AM" src="https://user-images.githubusercontent.com/25443836/141795229-b81bfe17-64cb-4a55-a292-86a961716f67.png">
Did you try using the "await" keyword in front of your dynamic import and calling it from within an async function? According to the official docs, that is one valid approach that can be taken. I have never tried to dynamically import a style file but I noticed that you aren't awaiting the import above.

https://nextjs.org/docs/advanced-features/dynamic-import

**Reply by joulev** [reply]
Thank you very much! It did take me a few tries but finally, to my surprise, it worked!

**Reply by ollieFrontendLdn** [reply]
Hi @joulev , out of interest how did you get this to work? I am working on something similar. Would you be able to post a snippet showing your dynamic import?

Thanks

**Reply by joulev** [reply]
Hi @ollieFrontendLdn, you can do like this for example,

```css
/* Home.blue.module.css */
.myClass {
  color: blue;
}
/* Home.red.module.css */
.myClass {
  color: red;
}
```

```js
import { useEffect, useState } from "react";

export default function Home() {
  const [isRed, setisRed] = useState(false);
  const [myClass, setMyClass] = useState("");
  useEffect(() => {
    (async () => {
      const theme = isRed
        ? await import("../styles/Home.red.module.css")
        : await import("../styles/Home.blue.module.css");
      setMyClass(theme.myClass);
    })();
  }, [isRed]);
  return (
    <div>
      <div className={myClass}>hello world {isRed ? "is red" : "is blue"}.</div>
      <button onClick={() => setisRed(isRed => !isRed)}>Toggle mode</button>
    </div>
  );
}
```

**Reply by ollieFrontendLdn** [reply]
Thanks @joulev

**Reply by addy-pathania** [reply]
how can i use variable inside the import function?
e.g import(`../styles/${theme}/Home.r.module.css`)

Nextjs throws an exception when trying to use a variable in the import function.

**Reply by joulev** [reply]
> how can i use variable inside the import function?
> 
> e.g import(`../styles/${theme}/Home.r.module.css`)
> 
> 
> 
> Nextjs throws an exception when trying to use a variable in the import function.

@sc-addypathania No, you can't use variables inside imports as far as I know.
